whatshap/cli/haplotagphase.py

```diff
@@ -113,18 +117,27 @@ def run_haplotagphase(
             sample_to_super_reads, sample_to_components = (dict(), dict())
             for sample in vcf_reader.samples:
                 logger.info(f"Processing sample {sample}")
+                genotypes = variant_table.genotypes_of(sample)
                 with timers("read-bam"):
-                    reads, _ = phased_input_reader.read(chromosome, variant_table.variants, sample)
+                    reads, _ = phased_input_reader.read(
+                        chromosome, variant_table.variants, sample, predicted_genotypes=genotypes
```

Comment:
"predicted" is accurate in the sense of "the genotype predicted by the variant caller", but this isn’t relevant here. Did you actually mean "restricted" (in the sense of "genotypes restricted to those observed in the current sample")? Suggestions: `only_genotypes`, `restrict_genotypes`. (I think anything with `valid` or `allowed` in the number gives the wrong impression that "not allowed" genotypes will raise an error.)